qemu/tests/curl_cookie_with_secret.py

```diff
@@ -0,0 +1,70 @@
+import os
+import signal
+import logging
+
+from virttest import utils_misc
+from virttest import qemu_storage
+from virttest import error_context
+
+from avocado.utils import process
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    1) Start tcpdump to capture the request
+    2) Access libcurl image by qemu-img
+    3) Wait till tcpdump finished
+    4) tcpdump should catch the cookie data
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def _get_tcpdump_pid(dump_file):
+        cmd = ("ps -ef|grep tcpdump|grep %s|grep -v grep|awk '{print $2}'"
+               % dump_file)
+        return process.system_output(cmd, shell=True,
+                                     ignore_status=True).strip()
+
+    def _wait_for_tcpdump_done(dump_file):
+        response_timeout = params.get_numeric('response_timeout', 10)
+        if not utils_misc.wait_for(lambda: not _get_tcpdump_pid(dump_file),
+                                   response_timeout, 0, 1):
+            test.fail('tcpdump is running unexpectedly')
+
+    def _cleanup(dump_file):
+        if os.path.exists(dump_file):
+            os.unlink(dump_file)
+
+        pid = _get_tcpdump_pid(dump_file)
+        if pid:
+            os.kill(int(pid), signal.SIGKILL)
+
+    tag = params['remote_image_tag']
+    img_params = params.object_params(tag)
+    img_obj = qemu_storage.QemuImg(img_params, None, tag)
+    dump_file = utils_misc.generate_tmp_file_name('%s_access_tcpdump' % tag,
+                                                  'out')
+
+    logging.info('start tcpdump, save packets in %s' % dump_file)
+    process.system(
+        params['tcpdump_cmd'].format(server=img_params['curl_server'],
+                                     dump_file=dump_file),
+        shell=True, ignore_status=True, ignore_bg_processes=True
+    )
```

Comment:
Yongxue, I am afraid we cannot get the pid if we use process.run(), unless we use process.SubProcess to create a process object, then we can call obj.get_pid or obj.kill, as it will cost a large code change, I will keep the current code unchanged, and will use process.SubProcess to run 'daemon-like' processes in future, thanks for pointing this out.

Reply:
Hi @zhencliu 
All right, got it.

Reply:
Hi @zhencliu 
BTW, the 
```
def run(cmd, timeout=None, verbose=True, ignore_status=False,
        allow_output_check=None, shell=False,
        env=None, sudo=False, ignore_bg_processes=False,
        encoding=None):
```
is call the `process.SubProcess`.

Code flow:
`def run(cmd, timeout=None, verbose=True...)` -> `klass = get_sub_process_klass(cmd)` -> `def get_sub_process_klass(cmd):` -> `return SubProcess`

But your solution also looks good to me.
Thanks.